tests/optim/utils/test_model_utils.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python3
+import unittest
+
+import torch
+import torch.nn.functional as F
+
+import captum.optim._utils.models as model_utils
+from captum.optim._models.inception_v1 import googlenet
+from tests.helpers.basic import BaseTest, assertTensorAlmostEqual
+
+
+class TestLocalResponseNormLayer(BaseTest):
+    def test_local_response_norm_layer(self) -> None:
+        size = 5
+        alpha = 9.999999747378752e-05
+        beta = 0.75
+        k = 1
+
+        x = torch.randn(32, 5, 24, 24)
+        lrn_layer = model_utils.LocalResponseNormLayer(
+            size=size, alpha=alpha, beta=beta, k=k
+        )
+
+        assertTensorAlmostEqual(
+            self,
+            lrn_layer(x),
+            F.local_response_norm(x, size=size, alpha=alpha, beta=beta, k=k),
+            0,
+        )
+
+
+class TestReluLayer(BaseTest):
+    def test_relu_layer(self) -> None:
+        x = torch.randn(1, 3, 4, 4)
+        relu_layer = model_utils.ReluLayer()
+        assertTensorAlmostEqual(self, relu_layer(x), F.relu(x), 0)
+
+
+class TestRedirectedReluLayer(BaseTest):
+    def test_forward_redirected_relu_layer(self) -> None:
+        x = torch.randn(1, 3, 4, 4)
+        layer = model_utils.RedirectedReluLayer()
+        assertTensorAlmostEqual(self, layer(x), F.relu(x), 0)
+
+    def test_backward_redirected_relu_layer(self) -> None:
+        t_grad_input, t_grad_output = [], []
+
+        def check_grad(self, grad_input, grad_output):
+            t_grad_input.append(grad_input[0].clone().detach())
+            t_grad_output.append(grad_output[0].clone().detach())
+
+        rr_layer = model_utils.RedirectedReluLayer()
+        x = (torch.randn(1, 3, 4, 4, requires_grad=True) - 5).clamp(0, 1)
+        rr_layer.register_backward_hook(check_grad)
+        rr_loss = rr_layer(x * 1).mean()
+        rr_loss.backward()
+
+        assertTensorAlmostEqual(self, t_grad_input[0], t_grad_output[0], 0)

```

Comment:
Why do we need t_grad_input and t_grad_output as arrays if we are using only the first element ?

Reply:
I guess at the time I had thought that I wouldn't be able to save the variables without doing that. But it looks like I can just define the variables as None before the function to save them.

Reply:
Arrays can be appended from inside the hook function whereas other variables cannot.